ding_dibs.email_reports.inc

```diff
@@ -0,0 +1,292 @@
+        $status = 'ok';
+        break;
+
+      case '100':
+        $status = 'warning';
+        break;
+
+      default:
+        $status = 'error';
+    }
+    $order_info_long = unserialize($record->order_info_long);
+    if (is_array($order_info_long) && count($order_info_long) > 0) {
+      $text = implode(array_shift($order_info_long)) . ' ';
+      $text .= implode(', ', array_map('array_shift', $order_info_long));
+    }
+    elseif (is_string($order_info_long)) {
+      $text = $order_info_long;
+    }
+    else {
+      $text = $record->order_info_short;
+    }
+    $row = array(
+      array(
+        'data' => check_plain($record->payment_order_id),
+        'class' => 'payment',
+      ),
+      array(
+        'data' => check_plain($record->payment_transaction_id),
+        'class' => 'payment',
+      ),
+      array(
+        'data' => check_plain($record->payment_time),
+        'class' => 'payment',
+      ),
+      array(
+        'data' => check_plain($record->customer_uid),
+        'class' => 'payment',
+      ),
+      array(
+        'data' => check_plain($text),
+        'class' => 'payment',
+      ),
+      array(
+        'data' => check_plain($record->payment_price),
+        'class' => 'payment',
+      ),
+      array(
+        'data' => $record->payment_status ? t('yes') : t('no'),
+        'class' => 'payment',
+      ),
+      array(
+        'data' => $record->provider_payment_status ? t('yes') : t('no'),
+        'class' => 'payment',
+      ),
+      array(
+        'data' => $record->dibs_capture_status ? t('yes') : t('no'),
+        'class' => 'payment',
+      ),
+      array(
+        'data' => '',
+        'class' => 'icon ' . $status,
+      ),
+    );
+    $rows[] = $row;
+    $no_transactions_msg = FALSE;
+  }
+
+  $output = theme('table', array(
+    'header' => $header,
+    'rows' => $rows,
+  ));
+  if ($no_transactions_msg) {
+    $output .= $no_transactions_msg;
+  }
+  return render($output);
+}
+
+/**
+ * Callback that builds page reports run cron.
+ *
+ * @return
+ *   Returns tabular representation of values results.
+ */
+function ding_dibs_cron_report() {
+  drupal_set_title(t('Ding DIBS cron report'));
+  $report = db_select('ding_dibs_report_cron', 'd')
+    ->fields('d', array('id', 'time_start', 'status', 'text'))
+    ->orderBy('id', 'DESC')
+    ->execute()
+    ->fetchAll();
+  $max_count = count($report);
+  $no_transactions_msg = '<p>' . t('No cron report.') . '</p>';
+  if (!$max_count) {
+    return $no_transactions_msg;
+  }
+  $rows = array();
+  drupal_add_css(drupal_get_path('module', 'ding_dibs') . '/css/ding_dibs.css');
+  $header = array(array('data' => t('Message'), 'class' => 'payment'));
+  foreach ($report as $record) {
+    if (empty($record->text)) {
+      $day = format_date($record->time_start, 'custom', 'd');
+      $hour = format_date($record->time_start, 'custom', 'H:i');
+      $record->text = $day . 'th september ' . $hour . ': Job failed.';
```

Comment:
This makes no sense. Why is September hardcoded here?

Please use the appropriate date/time output functions. Use localization here for better support for extracting translatable strings.